### DISCUSSION TITLE
release and ownership of the project

### DISCUSSION THREAD

**Discussion by oroulet** [locka99/opcua#306] [General]
Hi @locka99 ,

Looks like you are really busy currently and current release is broken with latest rust. May I propose that you create a github organization and give possibilities for others to release new versions?

I also happen to be the original author of the https://github.com/FreeOpcUa/opcua-asyncio and owner of https://github.com/FreeOpcUa so a possibility is also that I make you admin of that organization and to move your repository to that organization. It is not linked to any company, just people like you and me doing side work using UA. Having one organization for several language can give these project more visibility and make sure someone is able to maintain things when we are busy.

I am also  working a lot with rust  at work currently so me or my colleagues will most probably come up with PR to your https://github.com/locka99/opcua

Olviier

**Comment by oroulet** [comment]
@locka99 What do you think? If you do not answer I will probably fork the project and publish a new crate. that is a bit stupid, I would rather collaborate

**Comment by locka99** [comment]
I'm open to other devs helping out and being granted as maintainers but I would not be in favour of moving the repo. I am not responding to much personal communication at the moment because of Christmas and other commitments. If you have a need for an async client, then it would be better to look at the sync APIs and see if there is a way to produce an async version as analogous to it as possible but with awaits instead of synchronous calls. Personally I don't think async client is much of a big deal as an async server would be for obvious reasons - clients typically aren't connected to so many servers that its even an issue, where as a server might want to scale to hundreds of clients while limiting the thread pool size.

**Reply by oroulet** [reply]
Did I wrote anything about async client? 
My main "issue" currently is that current release is broken, it does not compile with latest rust. So I was proposing an improvement to process so you do not have to be in charge of everything. For my part moving repositories to organization has been very positive.
Everything else I hopefully can fix myself and send MR. And yes I have been wondering about async client, but that is some work and rust is quite good at mixing sync and async

**Reply by locka99** [reply]
If there is a specific breakage in the last released crate I can release a version from master after I finish some crate updates. Master does build fine on recent Rust compilers so there is an option to use a hash as a dependency from a cargo file.

**Reply by oroulet** [reply]
> If there is a specific breakage in the last released crate I can release a version from master after I finish some crate updates. 

yes bug is reported here:  https://github.com/locka99/opcua/issues/294

> Master does build fine on recent Rust compilers so there is an option to use a hash as a dependency from a cargo file

yes that is what everyone is doing I suppose...

**Comment by locka99** [comment]
I've released 0.12.0 today with upgraded packages for some things. So it's now 0.13.0 for master. I have a substantial pubsub branch I have on the side but I don't think its ready to merge in yet because it's still building foundations for that feature and needs to demonstrate something end to end before I land but I hope it might go in for 0.13.0

**Reply by oroulet** [reply]
pubsub is a big one! That is impressive.  We never managed to really agree on Python side so it is still not merged. It was hard and I did not have time.
Btw what is status of Rust opcua client? do you have some big things missing, apart from async API? Looks like the wiki is rather outdated..
